bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/ResponseBuilder.java

```diff
@@ -21,16 +21,21 @@
 package org.apache.bookkeeper.proto;
 
 import io.netty.buffer.ByteBuf;
+import org.apache.bookkeeper.util.ByteBufList;
 
 class ResponseBuilder {
     static BookieProtocol.Response buildErrorResponse(int errorCode, BookieProtocol.Request r) {
         if (r.getOpCode() == BookieProtocol.ADDENTRY) {
             return BookieProtocol.AddResponse.create(r.getProtocolVersion(), errorCode,
                                                   r.getLedgerId(), r.getEntryId());
-        } else {
-            assert(r.getOpCode() == BookieProtocol.READENTRY);
+        } else if (r.getOpCode() == BookieProtocol.READENTRY) {
             return new BookieProtocol.ReadResponse(r.getProtocolVersion(), errorCode,
                                                    r.getLedgerId(), r.getEntryId());
+        } else {
+            assert(r.getOpCode() == BookieProtocol.BATCH_READ_ENTRY);
+            return new BookieProtocol.BatchedReadResponse(r.getProtocolVersion(), errorCode,
+                    r.getLedgerId(), r.getEntryId(), ((BookieProtocol.BatchedReadRequest) r).getRequestId(),
+                    ByteBufList.get());
```

Comment:
Why pass an empty ByteBufList here

Reply:
Because it build error response, the reason is same with above, I will return null directly.